pkg/api/scylla/v1/types_cluster.go

```diff
@@ -331,6 +331,14 @@ type SchedulerTaskSpec struct {
 	// +kubebuilder:default:=3
 	// +optional
 	NumRetries *int64 `json:"numRetries,omitempty"`
+
+	// cron specifies the task schedule as a cron expression. It supports an extended syntax including @monthly, @weekly, @daily, @midnight, @hourly, @every X[h|m|s].
+	// +optional
+	Cron *string `json:"cron,omitempty"`
```

Comment:
is this mutually exclusive with start date? should you validate both are not set?

Reply:
I don't see sctool doing this. Looking at https://github.com/scylladb/scylla-manager/pull/3720#issuecomment-1956621668 and https://github.com/scylladb/scylla-manager/issues/3750 I'm not even sure I understand how they interact with each other. @Michal-Leszczynski can you advise?

Reply:
@karol-kokoszka please correct me if I'm wrong, but `cron` works with `startDate` in a way that the first `cron` activation will scheduled after the `startDate` (not necessarily on the `startDate` itself). So the `startDate` can be perceived as a date when the `cron` task is added to SM.

Reply:
> but cron works with startDate in a way that the first cron activation will scheduled after the startDate

Indeed.

Reply:
so to my understanding cron is mutually exclusive with interval, but both can work with start date?

Reply:
Added validation for mutual exclusivity of cron and interval. A bit awkward since interval is not a pointer and defaulted.

Reply:
yeah, I think even with the startDate it's weird if your `startDate` is say on Monday and cron says it should run on Fridays. Lets keep the validation tight as we can never go back, but loosening is possible if a use case is found that is more valuable then the conflict I pointed out here.

Reply:
I'll add the validation here, also sent a separate PR moving all existing validation to the validating webhook: https://github.com/scylladb/scylla-operator/pull/1868.
I also created a tracker for the other fields which won't be cover in these two PRs: https://github.com/scylladb/scylla-operator/issues/1869.

Reply:
done

Reply:
@tnozicka I wonder, should we also validate the extended cron format on our side? We don't control it, so if it changes upstream, we'll get out of sync

Reply:
I'd start with a strict validation of the format and loosen if we bump into something. Kube seems to use a cron library as well, so this might be a good start https://github.com/kubernetes/kubernetes/blob/e800d15/go.mod#L58C2-L58C27